.gitlab-ci.yml

```diff
@@ -0,0 +1,44 @@
+stages:
+  - All In One Script
+
+cache:
+  key: ${sha1sum package-lock.json}
+  paths:
+    - .npm/
+
+before_script:
+  - npm -v
+  - node -v
+  - npm ci --cache .npm --prefer-offline
+  - npm rebuild
+
+Single Test For Performance:
+  stage: All In One Script
+  interruptible: true
+  resource_group: docker-compose
+  script:
+    - docker -v
+    - docker-compose -v
+    - echo "rebuild docker-compose ONLY if config was changed"
+    - cmp --silent docker-compose.yml ~/.runner-cache/docker-compose.yml ||
+      bash -c 'docker-compose down
+      && docker-compose build
+      && docker-compose -f docker-compose.yml -f docker-compose.docker.yml up -d
+      && mkdir -p ~/.runner-cache
+      && rm -rf ~/.runner-cache/docker-compose.yml
+      && cp docker-compose.yml ~/.runner-cache/'
+    - bash -c 'while [[ "$(curl -s -o /dev/null -w ''%{http_code}'' 127.0.0.1:3000)" != "200" ]]; do
+      if [[ $i < 31 ]] ;
+      then i=$((i+1)) && sleep 5 && echo "slep... $i";
+      else exit 1; fi
+      ; done'
+    - npm run service-status
+    - npm run lint -- --quiet
+    - npm run test:unit -- --forceExit
+    - npm run test:i
```

Comment:
this is integration test .right. ?
why we have both `test:i ` and `test:integration:headless` at package.json ?

Reply:
please leave `test:integration:headless` and remove `test:i`

Reply:
Done